File: BubbleSort.rb
def bubble_sort(array)
    array_length = array.size
    return array if array_length <= 1
  
    loop do
      swapped = false
  
      (array_length-1).times do |i|
        if array[i] > array[i+1]
          array[i], array[i+1] = array[i+1], array[i]
          swapped = true
        end
      end
  
      break if not swapped
    end
  
    array
  end
  
  unsorted_array = Array.new(5) { rand(1...100) }
  p ("Unsorted array: #{unsorted_array}")
  p ("Sorted array: #{bubble_sort(unsorted_array)}")

File: BubbleSort.dart
import 'dart:math';

List<int> bubbleSort(List<int> list) {
  var retList = new List<int>.from(list);
  var tmp;
  var swapped = false;
  do {
    swapped = false;
    for (var i = 1; i < retList.length; i++) {
      if (retList[i - 1] > retList[i]) {
        tmp = retList[i - 1];
        retList[i - 1] = retList[i];
        retList[i] = tmp;
        swapped = true;
      }
    }
  } while (swapped);

  return retList;
}

void main() {
  final random = new Random();
  final min = 1;
  final max = 100;
  List<int> array = [];
  for (int i = 0; i < 5; i++) {
    array.add(min + random.nextInt(max - min));
  }
  print("Unsorted array: $array");
  List<int> sortedarray = bubbleSort(array);
  print("Sorted array: $sortedarray");
}